File: frontend/interfaces/Huesped.ts
export type TipoDocumento = 'DNI' | 'LE' | 'LC' | 'Pasaporte' | 'Otro' | '';
export type CondicionIVA = 
    | 'RESPONSABLE_INSCRIPTO' 
    | 'MONOTRIBUTO' 
    | 'CONSUMIDOR_FINAL' 
    | 'EXENTO';
export interface Huesped {
  id: number;
  nombre: string;
  apellido: string;
  tipo_documento: TipoDocumento;
  num_documento: string;
  email: string;
  nacionalidad: string;
  fecha_nacimiento?: string;
  telefono?: string;
  cuit?: number; 
  ocupacion?: string; 
  condicionIVA: CondicionIVA;
  direccion: Direccion;
}

export interface CriteriosBusquedaHuesped {
  apellido: string;
  nombre: string;
  tipo_documento: TipoDocumento | '';
  num_documento: string;
}
export interface Direccion {

  calle: string;
  numero: number;
  piso: number;
  departamento: string;
  codigoPostal: number;
  localidad: string;
  provincia: string;
  pais: string;
}

File: backend/src/main/java/tp/desarrollo/model/Huesped.java

package tp.desarrollo.model;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import tp.desarrollo.enums.CondicionIVA;
import tp.desarrollo.enums.TipoDocumento;
import jakarta.validation.constraints.*;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class Huesped extends Persona{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Pattern(regexp = "^[0-9\\-\\s]+$", message = "El teléfono solo puede contener números, espacios y guiones")
    @NotBlank(message = "El teléfono es obligatorio")
    private String telefono;
    @Email(message = "Formato de correo electrónico inválido")
    private String email;
    @NotBlank(message = "La ocupación es obligatoria")
    private String ocupacion;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CondicionIVA condicionIVA; 

    public Huesped(String telefono, String email, String ocupacion, CondicionIVA condicionIVA,
                   String apellido, String nombre, TipoDocumento tipo_documento, int num_documento,
                   int cuit, LocalDate fecha_nacimiento, Direccion direccion, String nacionalidad) {
        super(apellido, nombre, tipo_documento, num_documento, cuit, fecha_nacimiento, direccion, nacionalidad);
        this.telefono = telefono;
        this.email = email;
        this.ocupacion = ocupacion;
        this.condicionIVA = condicionIVA;
    }
    public Huesped() {
    }
    @Override
    public String toString() {
        return this.getApellido() + ", " + this.getNombre();
    }
    
    //GETTERS
    public String getTelefono(){
        return  this.telefono;
    }
    public String getEmail(){
        return  this.email; 
    }
    public String getOcupacion(){
        return  this.ocupacion;
    }
    public CondicionIVA getCondicionIVA() { return condicionIVA; }
    
    
    //SETTERS
    public void setTelefono(String telefono){
        this.telefono = telefono;
    }
    public void setEmail(String email){
        this.email = email;
    }
    public void setOcupacion(String ocupacion){
        this.ocupacion = ocupacion;
    }
    public void setCondicionIVA(CondicionIVA condicionIVA) {
        this.condicionIVA = condicionIVA;
    }
}
